### PR TITLE
Voice tutorial:  Add procedure on setting up esp32 for voice control

### PR DESCRIPTION
**Pull request by c0ffeeca7**
## Proposed change

Assist tutorials:
- Add procedure on sestting up esp32 for voice control

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration others are adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/projects/private-voice-assistant/voice_remote_esp32.markdown [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
+title: "Setting up an ESP32 to control Home Assistant"
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_esp32.markdown]: Should we make it more clickbaity?

```suggestion
title: "$13 voice remote that controls Home Assistant"
```

File: source/projects/private-voice-assistant/voice_remote_esp32.markdown [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+your smart home. Issue commands and get responses!
+
+## Required material
+
+* Latest version of Home Assistant installed on your Home Assistant device 
+* A controllable device in your Home Assistant network
+* [Home Assistant Cloud](https://www.nabucasa.com) or a manually configured [Assist Pipeline](/integrations/assist_pipeline)
+* The password to your 2.4&nbsp;GHz Wi-Fi network
+* Chrome (or a Chromium-based browser like Edge) on desktop (not Android/iOS) 
+* ESP32 board with a built-in speaker: [M5Stack ATOM Echo Development Kit](https://shop.m5stack.com/products/atom-echo-smart-speaker-dev-kit?ref=NabuCasa)
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_esp32.markdown]: Probably just mention the product right away
```suggestion
* [M5Stack ATOM Echo Development Kit](https://shop.m5stack.com/products/atom-echo-smart-speaker-dev-kit?ref=NabuCasa)
```

File: source/projects/private-voice-assistant/voice_remote_esp32.markdown [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+---
+
+This tutorial will guide you to turn an ESP32 (ATOM Echo by M5) into the
+world's most private voice assistant. Pick up the tiny device to talk to
+your smart home. Issue commands and get responses!
+
+## Required material
+
+* Latest version of Home Assistant installed on your Home Assistant device 
+* A controllable device in your Home Assistant network
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_esp32.markdown]: I would leave this one out, seems obvious? 
```suggestion
```

File: source/projects/private-voice-assistant/voice_remote_esp32.markdown [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+## Flashing the firmware onto the ATOM Echo
+
+1. Plug the USB-C cable into the M5Stack ATOM Echo Development Kit and connect it to your computer.
+1. Make sure this page is opened in a Chromium-based browser and select the **Connect** button below.
+
+   <script type="module" src="https://unpkg.com/esp-web-tools@9/dist/web/install-button.js?module"></script>
+   <esp-web-install-button manifest="https://firmware.esphome.io/voice-assistant/m5stack-atom-echo/manifest.json"></esp-web-install-button>
+
+   * In the pop-up window, select the USB serial port and select **Connect**.
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_esp32.markdown]: It's not clear sometimes what USB serial port belongs to the device. To make it obvious, let users plug in the device after the window is open, and they see the serial port appear. That's the one they should pick.

File: source/projects/private-voice-assistant/voice_remote_esp32.markdown [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        * Select **Install Voice Assistant**, then **Install**.
+        * Follow the instructions provided by the installation wizard.
+        * When prompted, enter the credentials to your 2.4&nbsp;GHz Wi-Fi network.
+          * As the **Network name**, enter the SSID of your 2.4&nbsp;GHz Wi-Fi.
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_esp32.markdown]: There should be a drop down to pick your network, unless you select "Other". It can sometimes be that the device doesn't have enough time to scan for networks.

File: source/projects/private-voice-assistant/voice_remote_esp32.markdown [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+          * The ATOM Echo now joined your network.
+1. In Home Assistant, go to {% my integrations title="**Settings** > **Devices & Services**" %}.
+   * You should now see a new **M5Stack Atom Echo** integration.
+   * Select **Configure** and then **Submit**.
+   * The ESP32 integration should now show all the entities.
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_esp32.markdown]:
```suggestion
   * The ESPHome config entry should now show all the entities.
```

File: source/projects/private-voice-assistant/voice_remote_esp32.markdown [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+   ![atom echo discovered](/images/assist/m5stack-atom-echo-discovered-03.png)
+1. Congratulations! You've setup your ATOM Echo device to voice control Home Assistant. Now give some commands.
+   * Press the button on your ATOM Echo.
+     * The LED should light up in blue.
+     * Say a [supported voice command](/docs/assist/builtin_sentences/). For example, *Turn off the light in the kitchen*.
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_esp32.markdown]: Just a note, that we default to the language configured in Home Assistant, which might not be English.

File: source/projects/private-voice-assistant/voice_remote_esp32.markdown [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+     * You can also ask a question, such as
+        * *Is the front door locked?*
+        * *Which lights are on in the living room?*
+        * Make sure you’re using the area name as you defined it in Home Assistant. If you have a room called bathroom, the phrase *Turn on the lights in the bath* won’t work.
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_esp32.markdown]: This seems a bit too specific to be listed here? Probably better in a generic voice command docs. Also, good to note in that case that we support area aliases to cover this.

File: source/projects/private-voice-assistant/voice_remote_esp32.markdown [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+   * Let go of the button.
+     * The LED should light up in green.
+     * Home Assistant will confirm the action.
+1. Your command was not recognized? Test if the command works:
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_esp32.markdown]: Should this be under troubleshooting?

**Review by frenck — DISMISSED** [review]
Looks good to me, couldn't fine anything out of the ordinary 👍 

../Frenck

File: source/_includes/asides/docs_navigation.html [hunk 10]
```diff
@@ -46,6 +46,10 @@ <h1 class="title delta">Topics</h1>
           <li>{% active_link /docs/assist/apple/ Assist for Apple devices %}</li>
           <li>{% active_link /docs/assist/builtin_sentences/ Built-in sentences %}</li>
           <li>{% active_link /docs/assist/custom_sentences/ Custom sentences %}</li>
+          <li>{% active_link /projects/private-voice-assistant/voice_remote_esp32/ Setting up a $13 voice remote to control Home Assistant %}</li>
+          <li>{% active_link /projects/private-voice-assistant/voice_remote_local_device/ Setting up a voice remote that processes locally %}</li>
+          <li>{% active_link /projects/private-voice-assistant/voice_remote_local_assistant/ Configuring a local assistant %}</li>
+          <li>{% active_link /projects/private-voice-assistant/voice_remote_expose_devices/ Exposing devices to your voice assistant %}</li>
```
Comment (balloob) [inline source/_includes/asides/docs_navigation.html]: These 2 should be under docs. They are not projects.

File: source/_includes/asides/docs_navigation.html [hunk 11]
```diff
@@ -46,6 +46,10 @@ <h1 class="title delta">Topics</h1>
           <li>{% active_link /docs/assist/apple/ Assist for Apple devices %}</li>
           <li>{% active_link /docs/assist/builtin_sentences/ Built-in sentences %}</li>
           <li>{% active_link /docs/assist/custom_sentences/ Custom sentences %}</li>
+          <li>{% active_link /projects/private-voice-assistant/voice_remote_esp32/ Setting up a $13 voice remote to control Home Assistant %}</li>
+          <li>{% active_link /projects/private-voice-assistant/voice_remote_local_device/ Setting up a voice remote that processes locally %}</li>
```
Comment (balloob) [inline source/_includes/asides/docs_navigation.html]: These 2 are projects but are linked in the docs navigation, should they just be under docs?

Also, the analog phone is the World's Most Private Voice Assistant. Currently it has 2 folders `private-voice-assistant/voice_remote_local_device/`. Let's change this to `/projects/worlds-most-private-voice-assistant`

The ESP tutorial is the "$13 voice remote for Home Assistant". It currently also has 2 folders. Let's change this to `/projects/13-usd-voice-remote`

File: source/projects/private-voice-assistant/voice_remote_esp32.markdown [hunk 12]
```diff
@@ -0,0 +1,81 @@
+---
+title: "Setting up a $13 voice remote to control Home Assistant"
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_esp32.markdown]:
```suggestion
title: "$13 voice remote to control Home Assistant"
```

File: source/projects/private-voice-assistant/voice_remote_esp32.markdown [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+---
+
+This tutorial will guide you to turn an ATOM Echo into the
+world's most private voice assistant. Pick up the tiny device to talk to
+your smart home. Issue commands and get responses!
+
+* Don't have Home Assistant Cloud? You just want fully local processing? Follow this procedure: [Setting up a fully local voice remote device](/projects/private-voice-assistant/voice_remote_local_device/)
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_esp32.markdown]: We should link this video instead:

```suggestion
<lite-youtube videoid="w6QxGdxVMJs" videotitle="$13 voice remote for Home Assistant
"></lite-youtube>
```

The local pipeline suggestion should be on the `assist_pipeline` docs, which will be linked in requirements.

File: source/projects/private-voice-assistant/voice_remote_esp32.markdown [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+
+## Required material
+
+* Latest version of Home Assistant installed on your Home Assistant device 
+* [Home Assistant Cloud](https://www.nabucasa.com)
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_esp32.markdown]:
```suggestion
* [Home Assistant Cloud](https://www.nabucasa.com) or a manually configured [Assist Pipeline](/integrations/assist_pipeline)
```

File: source/projects/private-voice-assistant/voice_remote_esp32.markdown [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+
+This tutorial will guide you to turn an ATOM Echo into the
+world's most private voice assistant. Pick up the tiny device to talk to
+your smart home. Issue commands and get responses!
+
+* Don't have Home Assistant Cloud? You just want fully local processing? Follow this procedure: [Setting up a fully local voice remote device](/projects/private-voice-assistant/voice_remote_local_device/)
+
+## Required material
+
+* Latest version of Home Assistant installed on your Home Assistant device
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_esp32.markdown]:
```suggestion
* Home Assistant 2023.5 or later
```

File: source/projects/private-voice-assistant/voice_remote_esp32.markdown [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+   ![Open My link](/images/assist/esp32-atom-flash-no-port.png)
+1. Select **Install Voice Assistant**, then **Install**.
+     * Follow the instructions provided by the installation wizard.
+     * Add the ATOM Echo to your Wi-Fi:
+       * When prompted, select your network from the list and enter the credentials to your 2.4&nbsp;GHz Wi-Fi network.
+       * Select **Connect**.
+       * The ATOM Echo now joined your network. Select **Add to Home Assistant**.
+1. This opens the **My** link to Home Assistant. 
+   * If you haven't used My links before and previously changed the hostname, in `http://homeassistant.local:8123` replace `homeassistant` with the hostname or IP address of your local Home Assistant instance.
+   * Select **Save**, then **Open link**.
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_esp32.markdown]: "Select Save" is only applicable if people haven't used My links before

File: source/projects/private-voice-assistant/voice_remote_expose_devices.markdown [hunk 17]
```diff
@@ -0,0 +1,15 @@
+---
+title: "Expose devices"
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_expose_devices.markdown]: This should not be under projects. This is generic documentation that applies to Assist, Google and Amazon. I suggest this is moved to it's own PR.

File: source/projects/private-voice-assistant/voice_remote_local_assistant.markdown [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+---
+
+If you have Home Assistant Cloud, the voice assistant uses Home Assistant Cloud by default.
+This is because the Cloud version currently still performs better than the local version.
+The performance depends a bit on the hardware you run it on. To set up a fully local, private voice assistant, follow these steps.
+
+## Setting up a local voice assistant
+
+1. Install the add-ons to convert text into speech and vice versa.
+   * Under **Settings** > **Add-ons** > **Z-Wave JS**, install the {% my supervisor_addon addon="whisper" title="**Whisper**" %} and the {% my supervisor_addon addon="piper" title="**Piper**" %}
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_local_assistant.markdown]: Z-Wave JS?

File: source/projects/private-voice-assistant/voice_remote_local_assistant.markdown [hunk 19]
```diff
@@ -0,0 +1,35 @@
+---
+title: "Configuring a local voice assistant"
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_local_assistant.markdown]: These instructions should be moved into it's own PR. They are also not related to the `projects` but should be generic Assist documentation page (or maybe live on the assist pipeline docs?)

File: source/projects/private-voice-assistant/voice_remote_local_assistant.markdown (hunk 18 above)
Comment (balloob) [reply source/projects/private-voice-assistant/voice_remote_local_assistant.markdown]: Also, you don't need to go to settings/add-ons if you click a My link, as that deeplinks to the right location.

File: source/projects/private-voice-assistant/voice_remote_local_assistant.markdown [hunk 20]
```diff
@@ -0,0 +1,35 @@
...
+The performance depends a bit on the hardware you run it on. To set up a fully local, private voice assistant, follow these steps.
+
+## Setting up a local voice assistant
+
+1. Install the add-ons to convert text into speech and vice versa.
+   * Under **Settings** > **Add-ons** > **Z-Wave JS**, install the {% my supervisor_addon addon="whisper" title="**Whisper**" %} and the {% my supervisor_addon addon="piper" title="**Piper**" %} 
+      ![Install the Whisper and Piper add-ons](/images/assist/piper-whisper-install-01.png)
+   * Start both add-ons. This may take a while.
+   * Once the add-ons are started, head over to the integrations under {% my integrations title="**Settings** > **Devices & Services**" %}.
+      * You should now see both integrations.
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_local_assistant.markdown]:
```suggestion
      * You should now see both being discovered by the [Wyoming integration](/integations/wyoming).
```

File: source/projects/private-voice-assistant/voice_remote_local_device.markdown [hunk 21]
```diff
@@ -0,0 +1,38 @@
+---
+title: "Setting up a fully local voice remote device"
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_local_device.markdown]: Let's remove this file. We have instructions on how to create a local pipeline and instructions on how to setup a device to talk to pipelines.

File: source/projects/private-voice-assistant/voice_remote_local_assistant.markdown (hunk 19 above)
Comment (balloob) [reply source/projects/private-voice-assistant/voice_remote_local_assistant.markdown]:
```suggestion
title: "Configuring a local assist pipeline"
```

File: source/projects/private-voice-assistant/voice_remote_local_assistant.markdown [hunk 22]
```diff
@@ -0,0 +1,35 @@
+---
+title: "Configuring a local voice assistant"
+---
+
+If you have Home Assistant Cloud, the voice assistant uses Home Assistant Cloud by default.
+This is because the Cloud version currently still performs better than the local version.
+The performance depends a bit on the hardware you run it on. To set up a fully local, private voice assistant, follow these steps.
```
Comment (balloob) [inline source/projects/private-voice-assistant/voice_remote_local_assistant.markdown]:
```suggestion
In Home Assistant, the Assist pipelines are made up of various components that together form a voice assistant.

For each component you can choose from different options. We have prepared a speech-to-text and text-to-speech option that runs fully local.

The speech-to-text option is [Whisper](https://github.com/openai/whisper). It's an open source AI model that supports [various languages](https://github.com/openai/whisper#available-models-and-languages). We use a forked version called [faster-whisper](https://github.com/guillaumekln/faster-whisper). On a Raspberry Pi 4, it takes around 8 seconds to process incoming voice commands. On an Intel NUC it is done in under a second.

For text-to-speech we have developed [Piper](https://github.com/rhasspy/piper). Piper is a fast, local neural text to speech system that sounds great and is optimized for the Raspberry Pi 4. It supports [many languages](https://rhasspy.github.io/piper-samples/). On a Raspberry Pi, using medium quality models, it can generate 1.6s of voice in a second.
```

File: source/projects/private-voice-assistant/voice_remote_esp32.markdown (hunk 12 above)
Comment (c0ffeeca7) [reply source/projects/private-voice-assistant/voice_remote_esp32.markdown]: I'm using a system here. The -ing form is a visual cue telling readers that this is a task topic. A procedure that tells them step-by-step how to do something. So if they are looking for a procedure they can ignore anything that does not start with -ing. 

But I get that you are probably viewing this more from a marketing perspective.

File: source/projects/private-voice-assistant/voice_remote_local_assistant.markdown (hunk 18 above)
Comment (c0ffeeca7) [reply source/projects/private-voice-assistant/voice_remote_local_assistant.markdown]: I know. I'm always mentioning how to get there because I used My links a lot. The problem was then (and still is) for me that this way I don't learn where things are located within HA. And if I then look for something while I'm not coming from the documentation I struggle to find things. 
This way I can click on the My link for convenience but I still see the path where this is taking me.

**Review by balloob — APPROVED** [review]
